File: strip_ultra.js
const fs = require('fs');
const path = require('path');

const filePath = path.join(__dirname, 'src', 'pages', 'refactor-technical-review', 'components', 'constants', 'Procesadores.ts');

try {
    let content = fs.readFileSync(filePath, 'utf8');

    // Remove the ' \/ Ultra X' or ' \/ Core X' string from 'nombre'
    content = content.replace(/nombre:\s*'Core (i\d)\s*\/\s*Ultra\s*\d',/g, "nombre: 'Core $1',");
    content = content.replace(/nombre:\s*'Core (i\d)\s*\/\s*Core\s*\d',/g, "nombre: 'Core $1',");

    // Remove "Core i9 / Ultra 9" without commas just in case
    content = content.replace(/'Core (i\d)\s*\/\s*Ultra\s*\d'/g, "'Core $1'");
    content = content.replace(/'Core (i\d)\s*\/\s*Core\s*\d'/g, "'Core $1'");

    // Specifically for id names, remove -ultraX
    content = content.replace(/-ultra\d/g, "");

    // Find and remove generations containing Ultra or Serie Core
    // We regex match the { id: ..., nombre: ..., modelos: [...] }
    const genRegex = /\{\s*id:\s*'[^']*',\s*nombre:\s*'(?:Ultra|Serie Core)[^']*',\s*(?:año:\s*\d+,\s*)?(?:arquitectura:\s*'[^']*',\s*)?modelos:\s*\[[\s\S]*?\]\s*\},?/g;
    
    content = content.replace(genRegex, '');
    
    // Clean up empty lines or double commas
    content = content.replace(/,\s*,/g, ',');
    content = content.replace(/\[\s*,/g, '[');

    fs.writeFileSync(filePath, content);
    console.log("Success");
} catch (error) {
    console.error("Error:", error);
}

File: strip_ultra.py
import re

filepath = r"src\pages\refactor-technical-review\components\constants\Procesadores.ts"

with open(filepath, 'r', encoding='utf-8') as f:
    text = f.read()

# 1. Clean the Family names: Core i9 / Ultra 9 -> Core i9
text = re.sub(r"nombre:\s*'Core (i\d)\s*/\s*Ultra\s*\d',", r"nombre: 'Core \1',", text)
text = re.sub(r"nombre:\s*'Core (i\d)\s*/\s*Core\s*\d',", r"nombre: 'Core \1',", text)

# 2. Find and remove specific generations containing Ultra or Core 3 100u
# The objects match the pattern { id: '...', ... modelos: [ ... ] }
# Since there are no nested braces inside 'modelos: []' other than { id, nombre }, 
# we can use a non-greedy regex to match the whole generation object:

gen_pattern = re.compile(
    r'\{\s*id:\s*\'(?:ultra[12]|core-2024)-[^\']+\',\s*nombre:.*?(?:Ultra|Serie Core).*?modelos:\s*\[.*?\]\s*\},?',
    re.DOTALL
)

# Apply replacement multiple times if needed (though DOTALL with non-greedy should work)
text = re.sub(gen_pattern, '', text)

# Just to be safe, there might be leading/trailing commas left in the code
# Let's clean up any empty spots or commas
text = re.sub(r',\s*,', ',', text)
text = re.sub(r'\[\s*,', '[', text)

with open(filepath, 'w', encoding='utf-8') as f:
    f.write(text)

print("Replacement complete.")
